src/Orleans.Core/Messaging/MessageSerializer.cs

```diff
@@ -320,12 +309,12 @@ private List<GrainAddress> ReadCacheInvalidationHeaders<TInput>(ref Reader<TInpu
         {
             if (value is null)
             {
-                writer.WriteVarInt32(-1);
+                writer.WriteVarUInt32(0);
                 return;
             }
 
             var numBytes = Encoding.UTF8.GetByteCount(value);
-            writer.WriteVarInt32(numBytes);
+            writer.WriteVarUInt32((uint)numBytes + 1);
```

Comment:
I assume this is to avoid the need for zigzag encoding the signed int? I wonder how much time this saves in practice.

Reply:
It's for a small size saving when encoding string lengths.